src/panels/lovelace/common/validate-condition.ts

```diff
@@ -134,6 +180,43 @@ export function checkConditionsMet(
   });
 }
 
+export function extractConditionEntityIds(
+  conditions: Condition[]
+): Set<string> {
+  const entityIds: Set<string> = new Set();
+  for (const condition of conditions) {
+    if (condition.condition === "numeric_state") {
+      if (
+        typeof condition.above === "string" &&
+        isValidEntityId(condition.above)
+      ) {
+        entityIds.add(condition.above);
+      }
+      if (
+        typeof condition.below === "string" &&
+        isValidEntityId(condition.below)
+      ) {
+        entityIds.add(condition.below);
+      }
+    } else if (condition.condition === "state") {
+      [
+        ...(ensureArray(condition.state) ?? []),
+        ...(ensureArray(condition.state_not) ?? []),
```

Comment:
`ensureArray` actually ensure that it returns an array (undefined tested) --> `??` shouldn't be needed here

https://github.com/hacf-fr/home-assistant-frontend/blob/ac5d576c5b3b052e7d5c726f687e027cd0f9fad9/src/common/array/ensure-array.ts#L6-L11

Reply:
It returns `undefined` is the value is `undefined`

Reply:
Oh yeah 🤦 
My bad